eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -45,17 +48,34 @@ foreach ($branch in $branches)
   $openPullRequests = $pullRequests | ? { $_.State -eq "open" }
   if ($openPullRequests.Count -gt 0)
   {
-    LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has open pull Requests. Skipping"
+    LogDebug "Branch [ $branchName ] in repo [ $RepoId ] has open pull Requests. Skipping"
     LogDebug $openPullRequests.url
     continue
   }
 
-  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request. Deleting Branch"
-  try{
-    Remove-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref ($branch.Remove(0,5)) -AuthToken $AuthToken
+  if ($LastCommitOlderThan) {
+    if (!$res.object -or !$res.object.url) {
+      LogWarning "No commit url returned from response. Skipping... "
+      continue
+    }
+    try {
+      $commitDate = Get-GithubReferenceCommitDate -commitUrl $res.object.url -AuthToken $AuthToken
+      if ($commitDate -and ($commitDate -gt $LastCommitOlderThan)) {
+        LogDebug "The branch $branch last commit date $commitDate is newer than the date $LastCommitOlderThan. Skipping."
+        continue
+      }
+    }
+    catch {
+      LogError "Get-GithubReferenceCommitDate failed with exception:`n$_"
+      exit 1
```

Comment:
We should watch to ensure this does not happen in any normal cases otherwise we might get to a spot where this clean-up continuously fail.